src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryFunction.java

```diff
@@ -198,17 +197,16 @@ public RepositoryDirectoryValue.Builder fetch(
                   RepositoryFetchProgress.ongoing(
                       RepositoryName.createUnvalidated(rule.getName()),
                       "fetch interrupted due to memory pressure; restarting."));
-          return fetch(rule, outputDirectory, directories, env, recordedInputValues, key);
+          yield fetch(rule, outputDirectory, directories, env, recordedInputValues, key);
         } finally {
           // At this point, the worker thread has definitely finished. But in some corner cases (see
           // b/330892334), a Skyframe restart might still happen; to ensure we're not tricked into
```

Comment:
Why does this condition cause an ExecutionException? I that in the case of b/330892334, the symptom would be that the worked thread returns with `Success(null)`, not an exception.

Reply:
why does which condition cause an ExecutionException? this is a `finally` branch.

Reply:
b/330892334 . TBH I don't know what I was thinking when I wrote this comment :(